types/node/index.d.ts

```diff
@@ -795,6 +795,14 @@ declare namespace NodeJS {
         disconnect(): void;
         connected: boolean;
 
+        /**
+         * The `process.allowedNodeEnvironmentFlags` property is a special,
+         * read-only `Set` of flags allowable within the [`NODE_OPTIONS`][]
+         * environment variable.
+         */
+        // TODO: This is readonly
+        allowedNodeEnvironmentFlags: Set<string>;
```

Comment:
> TODO

Maybe we can use this way?

`readonly allowedNodeEnvironmentFlags: Set<string>;`

Reply:
Eh, yea this technically readonly, the comment is about the actual `Set` tho.

Reply:
We could throw in `ReadonlySet`, `ReadonlyMap` and so on.

Basically just a `get` and `has` without the `set` and `add` functions of normal `Map`/`WeakMap` and `Set`/`WeakSet`s.

Reply:
I don’t think we should duplicate tbese inbuilt type definitions, probably better to wait for node typings to be bumbed up to a higher ts version